java/src/jmri/util/NonNullArrayList.java

```diff
@@ -0,0 +1,76 @@
+package jmri.util;
+import javax.annotation.Nonnull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+/**
+ * An ArrayList that FindBugs understands will never contain null elements.
+ *
+ * @see java.util.ArrayList
+ * @see java.util.List
+ * @author	Bob Jacobsen, Copyright (C) 2017
+ */
+public class NonNullArrayList<E> extends ArrayList<E> {
+
+    @Override
+    public boolean add(@Nonnull E e) { return super.add(e); }
```

Comment:
Do you want to throw an IllegalArgumentException here if e is null? Right now this accepts null in code, even though code checkers will note the issue.